Readability.js

```diff
@@ -1266,23 +1270,25 @@ Readability.prototype = {
     e = e || this._doc;
     if (!e)
       return;
-    var cur = e.firstChild;
 
-    // Remove any root styles, if we're able.
-    if (typeof e.removeAttribute === 'function' && e.className !== 'readability-styled')
-      e.removeAttribute('style');
-
-    // Go until there are no more child nodes
-    while (cur !== null) {
-      if (cur.nodeType === cur.ELEMENT_NODE) {
-        // Remove style attribute(s) :
-        if (cur.className !== "readability-styled")
-          cur.removeAttribute("style");
+    if (e.nodeType === e.ELEMENT_NODE &&
+        e.namespaceURI === 'http://www.w3.org/1999/xhtml' &&
```

Comment:
Because we're guaranteed element nodes, I don't think you need to check `e.nodeType`.

The namespaceURI check doesn't seem to work in JSDOMParser. I don't think it supports namespaces... But the attributes don't have meaning in SVG ( https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute ) and I don't think we'll have other namespaces mixed in, so I think you can just leave it out - or maybe do:

```js
if ((!e.namespaceURI || e.namespaceURI === "http://www.w3.org/1999/xhtml") &&
    e.className != "readability-styled")
```